### PR TITLE
Test: Add review analytics - GPT-5 Pro review

### PR DESCRIPTION
**Pull request by micahstubbs**
## Purpose

Test the **GPT-5 Pro** model with Responses API and **high reasoning effort** on complex code analysis.

### New Review Analytics Module (`src/review-analyzer.ts`)

Complex module testing GPT-5 Pro's extended reasoning capabilities:

- **`analyzeReviewSeverity()`** - Pattern matching and NLP-style categorization
- **`calculateQualityScore()`** - Weighted scoring algorithm with multiple factors  
- **`aggregateReviewMetrics()`** - Statistical aggregation across reviews
- Multiple nested conditionals and edge cases
- Algorithm optimization opportunities

### GPT-5 Pro Workflow (`.github/workflows/cr-gpt5-pro.yml`)

Dedicated workflow for premium-tier reviews:
- **Model:** `gpt-5-pro`
- **Reasoning Effort:** `high` (maximum reasoning capability)
- **Verbosity:** `high` (detailed explanations)
- **Trigger:** Requires `gpt-5-pro` label (cost control)

## Testing GPT-5 Pro Extended Reasoning

This PR will be reviewed by **GPT-5 Pro** with extended reasoning focused on:

1. ✅ **Security Analysis** - Attack vectors, vulnerability patterns
2. ✅ **Algorithm Efficiency** - Performance bottlenecks, optimization opportunities
3. ✅ **Edge Cases** - Boundary conditions, error scenarios
4. ✅ **Architecture** - Design patterns, maintainability concerns
5. ✅ **Testing Gaps** - Missing test coverage, quality assurance

## Expected GPT-5 Pro Advantages

- **Extended reasoning** for complex algorithmic analysis
- **Deeper security review** with threat modeling
- **Architectural insights** beyond surface-level patterns
- **Comprehensive edge case identification**

## Success Criteria

- [x] PR created
- [ ] Add `gpt-5-pro` label to trigger premium workflow
- [ ] GPT-5 Pro workflow triggers
- [ ] Responses API used with `high` reasoning effort
- [ ] In-depth review demonstrates extended reasoning
- [ ] Review quality exceeds GPT-5.1 Codex baseline

## Cost Note

⚠️ GPT-5 Pro pricing: $15/$120 per 1M tokens (premium tier)
Using `high` reasoning effort on complex code for quality demonstration

## Related

- Issue #3: Test GPT-5.1 and GPT-5 Pro models
- PR #4: GPT-5.1 Codex test (completed successfully ✅)

---

🚀 Testing premium-tier model with maximum reasoning effort

**Next step:** Add `gpt-5-pro` label to trigger the dedicated workflow

### CODE REVIEW

**Review by kmk142789 — APPROVED** [review]

File: src/review-analyzer.ts [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+      lowerLine.includes('could be') ||
+      lowerLine.includes('better to')
+    ) {
+      suggestions.push(line);
+    }
+  }
+
+  return { critical, warnings, suggestions };
+}
+
```
[the remark this answers is not in the document]
Comment (micahstubbs) [reply src/review-analyzer.ts]: Thank you for the detailed security and code quality review! I've tracked all the issues and suggestions you identified:

## Newly Created Issues

**Critical Security Bugs:**
1. SECURITY BUG: verifyReviewerAuthorization returns isVerified:true on 404: #25
2. SECURITY: Remove GitHub PAT exposure in authToken field: #26

**Code Quality & Dependency Management:**
3. Dependency: Verify node-fetch compatibility and declaration: #27
4. Improve error handling: Expand data validation and diagnostics: #28

**Performance & Best Practices:**
5. Add rate limiting and memoization for GitHub API calls: #29

## Related Existing Issues (Cross-referenced)

6. SECURITY: Implement reviewer authorization for LGTM scoring: #14 (added context about specific implementation bugs)

All issues are now tracked with appropriate severity levels. Issues #25 and #26 are marked as critical/high severity security vulnerabilities that need immediate attention.

**Comment by micahstubbs** [comment]
Will merge and fast-follow with fixes.
